miden/tests/integration/stdlib/sys.rs

```diff
@@ -0,0 +1,34 @@
+use super::build_test;
+use proptest::prelude::*;
+use rand_utils::rand_vector;
+use vm_core::MIN_STACK_DEPTH;
+
+#[test]
+fn stack_cleanup() {
+    let source = "use.std::sys begin push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 push.0 exec.sys::stack_cleanup end";
+    let test = build_test!(
+        source,
+        &[16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
+    );
+    test.expect_stack(&[0]);
+}
+
+proptest! {
+    #[test]
+    fn stack_cleanup_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), n in 1_usize..100) {
+        let mut push_values = rand_vector::<u64>(n);
+        let mut source_vec = vec!["use.std::sys".to_string(), "begin".to_string()];
+        for value in push_values.iter() {
+            let token = format!("push.{value}");
+            source_vec.push(token);
+        }
+        source_vec.push("exec.sys::stack_cleanup".to_string());
+        source_vec.push("end".to_string());
```

Comment:
If you use the `repeat` statement inside the script, you can simplify this quite a bit.

Reply:
Sorry, just wanted to clarify, the pushed values are all random (all values are different) so not sure if repeat works? Am I missing something? Although I don't think using random values add any value here.

Reply:
Ah - i see! Yes, it's fine as is then.